huaweicloud/resource_huaweicloud_rds_instance_v3_test.go

```diff
@@ -116,179 +123,160 @@ func testAccRdsInstanceV3_basic(val string) string {
-  cidr = "192.168.0.0/16"
-}
-	
-resource "huaweicloud_vpc_subnet_v1" "test" {
-  name          = "subnet-rds-test-%s"
-  cidr          = "192.168.0.0/16"
-  gateway_ip    = "192.168.0.1"
-  primary_dns   = "100.125.1.250"
-  secondary_dns = "100.125.21.250"
-  vpc_id        = huaweicloud_vpc_v1.test.id
-}
-
-resource "huaweicloud_networking_secgroup_v2" "secgroup_1" {
-  name = "sg-rds-test-%s"
-}
+%s
 
-resource "huaweicloud_rds_instance" "instance" {
-  name = "terraform_test_rds_instance%s"
-  flavor = "rds.pg.c2.medium"
-  availability_zone = ["%s"]
-  security_group_id = huaweicloud_networking_secgroup_v2.secgroup_1.id
-  subnet_id = huaweicloud_vpc_subnet_v1.test.id
-  vpc_id = huaweicloud_vpc_v1.test.id
+resource "huaweicloud_rds_instance" "test" {
+  name                  = "%s"
+  flavor                = "rds.pg.c6.large.2"
+  availability_zone     = [data.huaweicloud_availability_zones.test.names[0]]
+  security_group_id     = huaweicloud_networking_secgroup.test.id
+  subnet_id             = huaweicloud_vpc_subnet.test.id
+  vpc_id                = huaweicloud_vpc.test.id
   enterprise_project_id = "%s"
 
   db {
     password = "Huangwei!120521"
-    type = "PostgreSQL"
-    version = "10"
-    port = "8635"
+    type     = "PostgreSQL"
+    version  = "10"
+    port     = "8635"
   }
   volume {
     type = "ULTRAHIGH"
     size = 50
   }
   backup_strategy {
     start_time = "08:00-09:00"
-    keep_days = 1
+    keep_days  = 1
   }
 }
-	`, val, val, val, val, HW_AVAILABILITY_ZONE, HW_ENTERPRISE_PROJECT_ID_TEST)
+`, testAccRdsInstanceV3_base(val), val, HW_ENTERPRISE_PROJECT_ID_TEST)
 }
 
-func testAccCheckRdsInstanceV3Destroy(s *terraform.State) error {
-	config := testAccProvider.Meta().(*Config)
-	client, err := config.RdsV3Client(HW_REGION_NAME)
-	if err != nil {
-		return fmt.Errorf("Error creating sdk client, err=%s", err)
-	}
-
-	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "huaweicloud_rds_instance" {
-			continue
-		}
-
-		v, err := fetchRdsInstanceV3ByListOnTest(rs, client)
+func testAccCheckRdsInstanceV3Destroy(rName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+		client, err := config.RdsV3Client(HW_REGION_NAME)
 		if err != nil {
-			return err
+			return fmt.Errorf("Error creating huaweicloud rds client: %s", err)
 		}
-		if v != nil {
-			return fmt.Errorf("huaweicloud rds instance still exists")
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != rName {
+				continue
+			}
+
+			id := rs.Primary.ID
+			instance, err := getRdsInstanceByID(client, id)
+			if err != nil {
+				return err
+			}
+			if instance.Id != "" {
+				return fmt.Errorf("%s (%s) still exists", rName, id)
+			}
 		}
+		return nil
 	}
-
-	return nil
 }
 
-func testAccCheckRdsInstanceV3Exists() resource.TestCheckFunc {
+func testAccCheckRdsInstanceV3Exists(n string, instance *instances.RdsInstanceResponse) resource.TestCheckFunc {
```

Comment:
suggest move this func to front fo func testAccRdsInstanceV3_base for better code structure
we generally put terraform scripts at the end of code file, you can refer to other resources.

Reply:
Thanks, this function has been moved to a position in front of the terraform script.